src/libs/ApiUtils.js

```diff
@@ -0,0 +1,78 @@
+import lodashGet from 'lodash/get';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../ONYXKEYS';
+import CONFIG from '../CONFIG';
+import CONST from '../CONST';
+import * as Environment from './Environment/Environment';
+
+// To avoid rebuilding native apps, native apps use production config for both staging and prod
+// We use the async environment check because it works on all platforms
+let ENV_NAME = CONST.ENVIRONMENT.PRODUCTION;
+Environment.getEnvironment()
+    .then((envName) => {
+        ENV_NAME = envName;
+    });
+
+let stagingServerToggleState;
+Onyx.connect({
+    key: ONYXKEYS.USER,
+    callback: (val) => {
+        stagingServerToggleState = lodashGet(val, 'shouldUseStagingServer');
+    },
+});
+
+function shouldUseStagingServer() {
```

Comment:
JSDoc with `@returns {Boolean}` for this? 🙇

Reply:
This function has the same name as the property of the Onyx key. Can we rename it to something like `canUseStagingServer()`?

Reply:
ya in my mind it's a style guide thing and a code convention we use everywhere (not really something I want to debate in this PR so would recommend we just fix it).

https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#boolean-variables-and-props

tl;dr boolean variables have the "should" convention so a function is usually "can" instead of "should"